code/modules/codex/entries/mobs.dm

```diff
@@ -1,3 +1,67 @@
+/mob/living/carbon/Xenomorph/get_antag_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.antag_text)
+		return general_entry.antag_text
+
+/mob/living/carbon/Xenomorph/get_lore_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.lore_text)
+		return general_entry.lore_text
+
+/mob/living/carbon/Xenomorph/get_mechanics_info()
+	. = ..()
+	var/list/xeno_strings = list()
+
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.mechanics_text)
+		xeno_strings += general_entry.mechanics_text + "<br>"
+
+	xeno_strings += "<br><U>Basic Statistics for this Xeno are as follows</U>:"
+
+	xeno_strings += "Name: '[name]'"
+	xeno_strings += "Tier: [tier]"
+	xeno_strings += "Melee slash damage: between [xeno_caste.melee_damage_lower] and [xeno_caste.melee_damage_upper]"
+	xeno_strings += "Tackle damage: [xeno_caste.tackle_damage]"
+	if(mob_size)
```

Comment:
No need for the `if` check. `switch` will just skip it if there's no `mob_size`.
Consider it the same as a chain of `if else`, in its functionality.